app/containers/DappRequest/DappRequest.jsx

```diff
@@ -0,0 +1,51 @@
+// @flow
+import React, { useEffect, useMemo } from 'react'
+import { useWalletConnectWallet } from '@cityofzion/wallet-connect-sdk-wallet-react'
+
+import { ROUTES } from '../../core/constants'
+import InvokeFunction from '../../components/DappRequest/InvokeFunction'
+import VerifyMessage from '../../components/DappRequest/VerifyMessage'
+import SignMessage from '../../components/DappRequest/SignMessage'
+
+type Props = {
+  history: any,
+  showErrorNotification: ({ message: string }) => void,
+}
+
+const componentsByMethod: { [key: string]: any } = {
+  invokeFunction: InvokeFunction,
+  signMessage: SignMessage,
+  verifyMessage: VerifyMessage,
+}
+
+const DappRequest = ({ history, showErrorNotification }: Props) => {
+  const { request, session } = history.location.state
+
+  const { method } = request.params.request
+  const { rejectRequest } = useWalletConnectWallet()
+
+  const Component = useMemo(() => componentsByMethod[method], [method])
+
+  useEffect(
+    () => {
+      ;(async () => {
+        if (!Component) {
+          showErrorNotification({
+            message: 'unsupportedMethod',
+          })
+
+          await rejectRequest(request, {
+            code: 1,
+            message: 'unsupportedMethod',
+          })
+          history.push(ROUTES.DASHBOARD)
+        }
+      })()
+    },
+    [rejectRequest, Component, request, showErrorNotification, history],
+  )
```

Comment:
This is some pretty weird looking syntax, I think something like this would be cleaner:

```javascript
useEffect(() => {
  async function checkUnsupportedMethod() {
    if (!Component) {
      showErrorNotification({
        message: 'unsupportedMethod',
      });
      await rejectRequest(request, {
        code: 1,
        message: 'unsupportedMethod',
      });
      history.push(ROUTES.DASHBOARD);
    }
  };

  checkUnsupportedMethod();
}, [rejectRequest, Component, request, showErrorNotification, history]);
```

Reply:
Yea, I would even put the `checkUnsupportedMethod` on a separated `useCallback`